### PR TITLE
Ct 3526 migration ico appeal links

### PR DESCRIPTION
**Pull request by ymao2**
## Description

This PR is to fix/change the rake task for ico appeal link migration

## Self-review checklist

* [ ] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&modal=detail&selectedIssue=CT-3526

### CODE REVIEW

File: lib/tasks/relink_ico_to_foi_ir.rake [hunk 1]
```diff
@@ -22,15 +22,19 @@ namespace :ico_appeal do
...
-      ico_appeal_cases = Case::ICO::Base.all
+      ico_appeal_cases = Case::ICO::FOI.all
       ico_appeal_cases.each do | ico_appeal |
         puts "Checking ico appeal #{ico_appeal.number}"
         foi_ir_id = find_related_foi_ir_case_id(ico_appeal, ico_appeal_foi_ids_map)
         if foi_ir_id.present?
           foi_ir_case = Case::Base.find(foi_ir_id)
-          clear_up_old_links(ico_appeal, 'related')
+          existing_original_case_id = ico_appeal.original_case.id
+          clear_up_old_related_link(ico_appeal.id, foi_ir_id)
```
Comment (willmcb) [inline lib/tasks/relink_ico_to_foi_ir.rake:32]: Should this be plural? `clear_up_old_related_links` or is it only one link now? :)

File: lib/tasks/relink_ico_to_foi_ir.rake [hunk 2]
```diff
@@ -161,8 +165,15 @@ namespace :ico_appeal do
...
-      new_original_case_link =LinkedCase.new(case_id: ico_appeal.id, linked_case_id: foi_ir_case.id, type: :original)
+    def clear_up_old_related_link(ico_appeal_id, foi_ir_id)
+      case_linkage_records = LinkedCase.where(type: 'related', case_id: ico_appeal_id, linked_case_id: foi_ir_id)
+      case_linkage_records.each do | case_linkage_record |
+        case_linkage_record.destroy()
+      end
+    end
+
+    def create_new_case_link(ico_appeal, link_case_id, link_type)
+      new_original_case_link =LinkedCase.new(case_id: ico_appeal.id, linked_case_id: link_case_id, type: link_type.to_sym)
```
Comment (willmcb) [inline lib/tasks/relink_ico_to_foi_ir.rake:176]: Might want to add a space here 👍

**Review by willmcb — APPROVED** [review]
LGTM - two small comments you might want to look at 👍

File: lib/tasks/relink_ico_to_foi_ir.rake (hunk 1 above)
Comment (ymao2) [reply lib/tasks/relink_ico_to_foi_ir.rake:32]: @willmcb, related links can be more than one, but for this occasion, I clear up specific related link based on the linked_case_id. so it is only one link

**Pull request merged**